i18n-helpers/src/directives.rs

```diff
@@ -0,0 +1,130 @@
+use regex::{escape, Regex};
+
+#[derive(Debug, PartialEq)]
+pub enum Directive {
+    Skip,
+    TranslatorComment(String),
+}
+
+pub struct DirectiveFinder {
+    pattern_re: Regex,
+}
+
+impl Default for DirectiveFinder {
+    fn default() -> Self {
+        Self::new("i18")
+    }
+}
+
+impl DirectiveFinder {
+    pub fn new(prefix: &str) -> Self {
+        let pattern: String = r"<!-{2,}\s*".to_owned()
+            + &escape(prefix)
+            + &r"\s*[:-]"
+            + &r"(.*[^-])"  // the command part of the prefix, which must not end in a hyphen.
+            + &r"-{2,}>";
```

Comment:
Please put this into a single `format!` call instead. That should cause fewer allocations and also make it clearer that this is one big string. You might want to use the ["verbose" mode](https://docs.rs/regex/latest/regex/#example-verbose-mode) if that makes things clearer.